agent/dnsx_agent.py

```diff
@@ -42,40 +40,71 @@ def process(self, message: m.Message) -> None:
         if results is not None:
             self._emit_results(domain, results)
 
-    def _emit_results(self, domain: str, results: Dict) -> None:
+        if wordlist is not None:
+            results = self._run_dnsx(domain, wordlist)
+            if results is not None:
+                self._emit_results(domain, results)
+
+    def _emit_results(self, domain: str, results: List) -> None:
         """Parses results and emits records."""
         for record in result_parser.parse_results(results):
             logger.info('emitting result for %s', record)
             self.emit(selector='v3.asset.domain_name.dns_record',
                       data={'name': domain, 'record': record.record, 'values': record.value})
 
-    def _run_dnsx(self, domain: str):
+    def _run_dnsx(self, domain: str, wordlist: Optional[str] = None):
         """Run dnsx and returns the results."""
-        with tempfile.NamedTemporaryFile() as input_domain,\
-                tempfile.NamedTemporaryFile() as output_domain:
+        command = self._prepare_command(domain, wordlist)
+        logger.info('running command %s', command)
+        result = subprocess.run(command, capture_output=True, check=False)
+        if result.returncode == 0 and result.stdout != b'':
+            return [json.loads(l) for l in result.stdout.decode().split('\n') if l != '']
+        else:
+            logger.warning('Empty result file for domain %s', domain)
+
+    def _prepare_command(self, domain, wordlist: Optional[str]) -> List[str]:
+        """Prepare dnsx command."""
+        command = ['dnsx',
+                   '-silent',
+                   '-a', '-aaaa', '-cname', '-ns', '-txt', '-ptr', '-mx', '-soa', '-resp',
+                   '-json',
+                   '-d', domain]
+
+        if wordlist is not None:
+            command.extend(['-w', wordlist])
+        return command
+
+    def _run_dnsx_resolve(self, domain: str):
+        """Run dnsx and returns the results."""
+        with tempfile.NamedTemporaryFile() as input_domain:
             input_domain.write(domain.encode())
             input_domain.flush()
-            command = self._prepare_command(str(pathlib.Path(input_domain.name)), str(pathlib.Path(output_domain.name)))
+            command = self._prepare_command_resolve(input_domain.name)
             logger.info('running command %s', command)
-            subprocess.run(command, check=False)
-            try:
-                return json.load(output_domain)
-            except json.JSONDecodeError:
+            result = subprocess.run(command, capture_output=True, check=False)
+            if result.returncode == 0 and result.stdout != b'':
+                return [json.loads(l) for l in result.stdout.decode().split('\n') if l != '']
+            else:
                 logger.warning('Empty result file for domain %s', domain)
 
-    def _prepare_command(self, domain_file, output) -> List[str]:
+    def _prepare_command_resolve(self, domain_file) -> List[str]:
```

Comment:
This function is similar to _prepare_command, you use the signature _prepare_command(self, domain Optional[str], domain_file Optional[str],  wordlist: Optional[str]) and check

if domain is not None:
  command.extend
elif domain_file is not None:
command.extend